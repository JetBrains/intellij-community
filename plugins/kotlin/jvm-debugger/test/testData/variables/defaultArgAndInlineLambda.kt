// Covers KT-18574
// Should be updated when KT-79835 is fixed

// SHOW_KOTLIN_VARIABLES

inline fun <X> inlineXU(
    p: X, f: (X) -> Unit = {
        //Breakpoint!
        println(p.toString())
    }
) {
    //Breakpoint!
    return f(p)
}

fun main() {
    inlineXU("abc")
}