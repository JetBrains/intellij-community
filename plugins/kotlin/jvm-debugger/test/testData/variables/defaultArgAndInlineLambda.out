LineBreakpoint created at defaultArgAndInlineLambda.kt:9
LineBreakpoint created at defaultArgAndInlineLambda.kt:13
Run Java
Connected to the target VM
defaultArgAndInlineLambda.kt:13
InlineStackFrame inlineXU (defaultArgAndInlineLambda.kt:13)
    JavaValue[local] p: java.lang.Object = abc (defaultArgAndInlineLambda.kt:7)
defaultArgAndInlineLambda.kt:9
InlineStackFrame lambda 'inlineXU' in 'inlineXU' (defaultArgAndInlineLambda.kt:9)
    JavaValue[local] it: java.lang.Object = abc (defaultArgAndInlineLambda.kt:7)
Disconnected from the target VM

Process finished with exit code 0
